src/test/java/seedu/address/model/delivery/DeliveryListTest.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Test
+    public void setDeliveries_nullDeliveryList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> VALID_DELIVERY_LIST.setDeliveries((DeliveryList) null));
+        assertThrows(NullPointerException.class, () -> VALID_DELIVERY_LIST.setDeliveries((List<Delivery>) null));
+    }
+
+    @Test
+    public void sortByAddress_success() {
+        VALID_DELIVERY_LIST_ADDRESS_SORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_ADDRESS_SORTED.add(VALID_DELIVERY_1);
+        VALID_DELIVERY_LIST_ADDRESS_SORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_ADDRESS_SORTED.add(VALID_DELIVERY_3);
+
+        VALID_DELIVERY_LIST_ADDRESS_UNSORTED.add(VALID_DELIVERY_1);
+        VALID_DELIVERY_LIST_ADDRESS_UNSORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_ADDRESS_UNSORTED.add(VALID_DELIVERY_3);
+        VALID_DELIVERY_LIST_ADDRESS_UNSORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_ADDRESS_UNSORTED.sortByAddress();
+
+        assertArrayEquals(VALID_DELIVERY_LIST_ADDRESS_SORTED.asUnmodifiableObservableList().toArray(),
+                VALID_DELIVERY_LIST_ADDRESS_UNSORTED.asUnmodifiableObservableList().toArray());
+
+    }
+
+    @Test
+    public void sortByCost_success() {
+        VALID_DELIVERY_LIST_COST_SORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_COST_SORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_COST_SORTED.add(VALID_DELIVERY_3);
+        VALID_DELIVERY_LIST_COST_SORTED.add(VALID_DELIVERY_1);
+
+        VALID_DELIVERY_LIST_COST_UNSORTED.add(VALID_DELIVERY_1);
+        VALID_DELIVERY_LIST_COST_UNSORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_COST_UNSORTED.add(VALID_DELIVERY_3);
+        VALID_DELIVERY_LIST_COST_UNSORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_COST_UNSORTED.sortByCost();
+
+        assertArrayEquals(VALID_DELIVERY_LIST_COST_SORTED.asUnmodifiableObservableList().toArray(),
+                VALID_DELIVERY_LIST_COST_UNSORTED.asUnmodifiableObservableList().toArray());
+
+    }
+
+    @Test
+    public void sortByEta_success() {
+        VALID_DELIVERY_LIST_ETA_SORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_ETA_SORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_ETA_SORTED.add(VALID_DELIVERY_1);
+        VALID_DELIVERY_LIST_ETA_SORTED.add(VALID_DELIVERY_3);
+
+
+        VALID_DELIVERY_LIST_ETA_UNSORTED.add(VALID_DELIVERY_1);
+        VALID_DELIVERY_LIST_ETA_UNSORTED.add(VALID_DELIVERY_2);
+        VALID_DELIVERY_LIST_ETA_UNSORTED.add(VALID_DELIVERY_3);
+        VALID_DELIVERY_LIST_ETA_UNSORTED.add(VALID_DELIVERY_4);
+        VALID_DELIVERY_LIST_ETA_UNSORTED.sortByEta();
+
+        assertArrayEquals(VALID_DELIVERY_LIST_ETA_SORTED.asUnmodifiableObservableList().toArray(),
+                VALID_DELIVERY_LIST_ETA_UNSORTED.asUnmodifiableObservableList().toArray());
+
+    }
+
+    @Test
+    public void sortByStatus_success() {
```

Comment:
Nice test cases! Looking forward to test cases for the SortCommand in your next PR. Might be able to reuse some code here